x/tally/types/filters.go

```diff
@@ -8,52 +8,51 @@ import (
 	"golang.org/x/exp/constraints"
 )
 
+var (
+	_ Filter = &FilterNone{}
+	_ Filter = &FilterMode{}
+	_ Filter = &FilterStdDev{}
+)
+
 type Filter interface {
 	// ApplyFilter takes in a list of reveals and returns an outlier
 	// list, whose value at index i indicates whether i-th reveal is
-	// an outlier. Value of 1 indicates an outlier, and value of 0
-	// indicates a non-outlier reveal.
-	ApplyFilter(reveals []RevealBody) ([]int, error)
+	// an outlier, and a boolean indicating whether consensus in reveal
+	// data has been reached.
+	ApplyFilter(reveals []RevealBody, errors []bool) ([]int, bool)
+	// GasCost returns the cost of the filter in terms of gas amount.
+	GasCost() uint64
 }
 
-type FilterNone struct{}
+type FilterNone struct {
+	gasCost uint64
+}
 
 // NewFilterNone constructs a new FilterNone object.
-func NewFilterNone(_ []byte) (FilterNone, error) {
-	return FilterNone{}, nil
+func NewFilterNone(gasCost uint64) FilterNone {
+	return FilterNone{gasCost: gasCost}
 }
 
-// FilterNone declares all reveals as non-outliers, unless reveals are
-// empty or corrupt.
-func (f FilterNone) ApplyFilter(reveals []RevealBody) ([]int, error) {
-	if len(reveals) == 0 {
-		return nil, ErrEmptyReveals
-	}
-
-	var corruptCount int
-	for _, r := range reveals {
-		if r.ExitCode != 0 {
-			corruptCount++
-			continue
-		}
-	}
-	if corruptCount*3 > len(reveals) {
-		return nil, ErrCorruptReveals
-	}
+// FilterNone declares all reveals as non-outliers.
+func (f FilterNone) ApplyFilter(reveals []RevealBody, _ []bool) ([]int, bool) {
+	return make([]int, len(reveals)), true
+}
 
-	return make([]int, len(reveals)), nil
+func (f FilterNone) GasCost() uint64 {
+	return f.gasCost
 }
 
 type FilterMode struct {
 	dataPath string // JSON path to reveal data
+	gasCost  uint64
 }
 
 // NewFilterMode constructs a new FilerMode object given a filter
 // input.
 // Mode filter input looks as follows:
 // 0             1                  9       9+data_path_length
 // | filter_type | data_path_length |   data_path   |
-func NewFilterMode(input []byte) (FilterMode, error) {
+func NewFilterMode(input []byte, gasCostMultiplier uint64, replicationFactor uint16) (FilterMode, error) {
```

Comment:
We could consider storing the replication factor on the filter, since we're passing that data anyway to determine the gas cost. That way we can 'simply' refer to the replication factor wherever we did `len(reveals)` and that should also simplify handling missing reveals? 🤔

Reply:
I will address this in a separate PR as a part of #453 if that makes sense.